sdk/eventhub/azure-eventhub/azure/eventhub/_pyamqp/aio/_sasl_async.py

```diff
@@ -100,3 +100,20 @@ async def negotiate(self):
             return
         else:
             raise ValueError("SASL negotiation failed.\nOutcome: {}\nDetails: {}".format(*fields))
+
+class SASLWithWebSocket(SASLTransport):
```

Comment:
What's different about this class than the base SASLTransport class other than the construction of the WebSocketTransport?

Reply:
For the async version - nothing. Only the constructor is different

Reply:
In that case I don't think there's a reason to create a new class for this?

Reply:
the constructors are quite different and will involve too much unnecessary code - i think a better idea would be to put negotiate in a mixin and then use different constructors.

Reply:
actually, updated this part a little bit - we don't need an async transport super class anymore